geopyspark/geotrellis/catalog.py

```diff
@@ -453,3 +458,94 @@ def write(uri,
                                     tiled_raster_layer.srdd,
                                     TimeUnit(time_unit).value,
                                     IndexingMethod(index_strategy).value)
+
+
+class Attributes(object):
```

Comment:
Actually, I'm not sure if this would work as being private. Since you need to access the `read`, `write`, and `delete` methods, it seems odd to be working with a class that's supposed to be hidden. I guess it makes sense then to have it be inside the `AttributeStore` since what we're really controlling is the creation of `Attributes`.

If we don't want the user to have any access to `Attributes` then we'd have to move some of `Attributes`'s methods over to `AttributeStore` and have some kind caching where we store `(layer_name, zoom) -> Attributes` in the `AttributeStore` instance.